### PR TITLE
fix!: don't throw on index creation

### PR DESCRIPTION
**Pull request by NathanBhanji**
Refactors the adapter to throw on creating the collection but not on index creation, the indexes should be non blocking as it might not be as performant but it's not a critical failure. 

Also adds an error wrapper, I shortened some error messages whilst I was there.

### CODE REVIEW

File: src/adapter.ts [hunk 1]
```diff
@@ -272,7 +284,7 @@ export class MongoAdapter
     await this.getCollection().deleteMany(line);
   }
 
-  public async createDBIndex() {
```
Comment (incognitojam) [inline src/adapter.ts:275]: is this a breaking change?

Comment (NathanBhanji) [reply src/adapter.ts:275]: OOP yeah

**Review by incognitojam — APPROVED** [review]
I like the error wrapping stuff, and I tried running your tests but commenting out things and they did fail correctly so looks good.

File: src/adapter.ts (hunk 1 above)
Comment (NathanBhanji) [reply src/adapter.ts:275]: made it fix!: for a breaking change

**Pull request merged**